core/src/main/java/org/apache/calcite/sql/validate/LambdaScope.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.sql.validate;
+
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.StructKind;
+import org.apache.calcite.sql.SqlIdentifier;
+import org.apache.calcite.sql.SqlLambda;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.type.SqlTypeFamily;
+import org.apache.calcite.sql.type.SqlTypeName;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * The name-resolution context for expression inside a LAMBDA clause. The objects visible are the
+ * lambda paramters, and those inherited from the parent scope.
+ *
+ * <p>Consider "SELECT lambdaExp((x)-&gt;A.a+x) FROM A
+ * resolved in the lambda scope for "Lambda Parameters and Table columns of A"</p>
+ */
+public class LambdaScope extends ListScope {
+
```

Comment:
Maybe we can extend the `ParameterScope` to support parent scope instead a new `LambdaScope`.

Reply:
ParameterScope has a limited scope where as LambdaScope can access parent scope as well. It didn't seem right to me.